modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/discount/test/CommerceDiscountExpirationTest.java

```diff
@@ -0,0 +1,116 @@
+import com.liferay.petra.function.UnsafeRunnable;
+import com.liferay.portal.kernel.scheduler.SchedulerJobConfiguration;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.test.util.ServiceContextTestUtil;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerMethodTestRule;
+
+import java.math.BigDecimal;
+
+import java.util.Calendar;
+
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Balazs Breier
+ */
+@RunWith(Arquillian.class)
+public class CommerceDiscountExpirationTest {
+
+	@ClassRule
+	@Rule
+	public static AggregateTestRule aggregateTestRule = new AggregateTestRule(
+		new LiferayIntegrationTestRule(),
+		PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Test
+	public void testViewExpiredDiscountEntry() throws Exception {
+		CommerceDiscount commerceDiscount1 =
+			_commerceDiscountLocalService.addCommerceDiscount(
+				TestPropsValues.getUserId(), RandomTestUtil.randomString(),
+				CommerceDiscountConstants.TARGET_PRODUCT_GROUPS, false, null,
+				true, BigDecimal.ONE, CommerceDiscountConstants.LEVEL_L1,
+				BigDecimal.ONE, BigDecimal.ZERO, BigDecimal.ZERO,
+				BigDecimal.ZERO, CommerceDiscountConstants.TYPE_PERCENTAGE, 0,
+				false, true, 1, 1, 2020, 1, 1, 1, 1, 2020, 1, 1, false,
+				ServiceContextTestUtil.getServiceContext());
+
+		CommerceDiscount commerceDiscount2 =
+			_commerceDiscountLocalService.addCommerceDiscount(
+				TestPropsValues.getUserId(), RandomTestUtil.randomString(),
+				CommerceDiscountConstants.TARGET_PRODUCT_GROUPS, false, null,
+				true, BigDecimal.ONE, CommerceDiscountConstants.LEVEL_L1,
+				BigDecimal.ONE, BigDecimal.ZERO, BigDecimal.ZERO,
+				BigDecimal.ZERO, CommerceDiscountConstants.TYPE_PERCENTAGE, 0,
+				false, true, 1, 1, 2020, 1, 1, 1, 1, 2020, 1, 1, false,
+				ServiceContextTestUtil.getServiceContext());
+
+		commerceDiscount1 = _commerceDiscountLocalService.getCommerceDiscount(
+			commerceDiscount1.getCommerceDiscountId());
+
+		Assert.assertFalse(commerceDiscount1.isExpired());
+
+		commerceDiscount1.setExpirationDate(RandomTestUtil.nextDate());
+
+		commerceDiscount1 =
+			_commerceDiscountLocalService.updateCommerceDiscount(
+				commerceDiscount1);
+
+		commerceDiscount2 = _commerceDiscountLocalService.getCommerceDiscount(
+			commerceDiscount2.getCommerceDiscountId());
+
+		Assert.assertFalse(commerceDiscount2.isExpired());
+
+		Calendar calendar = Calendar.getInstance();
+
+		calendar.add(Calendar.DAY_OF_MONTH, 30);
+
+		commerceDiscount2.setExpirationDate(calendar.getTime());
+
+		commerceDiscount2 =
+			_commerceDiscountLocalService.updateCommerceDiscount(
+				commerceDiscount2);
```

Comment:
@matyaswollner @smottal I saw that you both mentioned this. This was my first thought too but it does not work. It simply ignores what's given at creation. I tried multiple dates, etc.

Reply:
On the other hand, this seems to be a bug or sloppy business logic.

Reply:
did you try also with "future" date? because there data you're passing in the add method is in the past and if I remember correctly is ignored if in the past

Reply:
yep, tried with future date, past date, present date.